testnet/stacks-node/src/neon_node.rs

```diff
@@ -914,10 +930,13 @@ fn spawn_miner_relayer(
 
                                     // proceed to mine microblocks
                                     debug!(
-                                        "Microblock miner tip is now {}/{}",
-                                        &consensus_hash, &block_header_hash
+                                        "Microblock miner tip is now {}/{} ({})",
+                                        &consensus_hash, &block_header_hash, StacksBlockHeader::make_index_block_hash(&consensus_hash, &block_header_hash)
                                     );
                                     miner_tip = Some((ch, bh, microblock_privkey));
+
+                                    Relayer::refresh_unconfirmed(&mut chainstate, &mut sortdb);
```

Comment:
I'm confused why we'd want to call `refresh_unconfirmed` (and also `send_unconfirmed_txs` for `RelayerDirective::ProcessTenure` - could you explain?

Reply:
This code path is invoked when the node wins the block race.  This will invalidate all unconfirmed microblocks, because the canonical Stacks chain tip will have changed (and the node only tracks unconfirmed microblocks for the canonical tip).  Therefore, we need to call `refresh_unconfirmed()` in order to reset the unconfirmed microblock state.  This in turn has the intended effect of changing which transactions are represented in the unconfirmed microblock stream (via `inner_process_tenure()`, which _may_ process _multiple_ Stacks blocks), so we must then call `send_unconfirmed_txs()` to synchronize them with the p2p thread.